File: Arrays/80. Remove Duplicates from Sorted Array II/80. Remove Duplicates from Sorted Array II.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
  let count = 1;
  let j = 1;

  for (let i = 1; i < nums.length; i++) {
    if (nums[i] === nums[i - 1]) count++;
    else count = 1;

    if (count <= 2) {
      nums[j] = nums[i];
      j++;
    }
  }

  return j;
};

// Time: O(n)
// Space: O(1)

File: 80. Remove Duplicates from Sorted Array II.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        count = 0
        j = 1

        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1]:
                count += 1
            else:
                count = 0

            if count < 2:
                nums[j] = nums[i]
                j += 1

        return j
